pkg/cmd/kafka/create/create.go

```diff
@@ -579,48 +574,84 @@ func promptKafkaPayload(opts *options, constants *remote.DynamicServiceConstants
 	if answers.BillingModel == accountmgmtutil.QuotaMarketplaceType {
-		validMarketPlaces := FetchValidMarketplaces(orgQuota.MarketplaceQuotas, answers.CloudProvider)
-		if len(validMarketPlaces) == 1 {
-			answers.Marketplace = validMarketPlaces[0]
-		} else {
-			marketplacePrompt := &survey.Select{
-				Message: f.Localizer.MustLocalize("kafka.create.input.marketplace.message"),
-				Options: validMarketPlaces,
-			}
-			err = survey.AskOne(marketplacePrompt, &answers.Marketplace)
-			if err != nil {
-				return nil, err
-			}
+		answers, err = marketplaceQuotaPrompt(orgQuota, answers, f, err)
+		if err != nil {
+			return nil, err
 		}
+	}
 
-		if len(validMarketPlaces) > 0 {
+	var sizes []kafkamgmtclient.SupportedKafkaSize
+	payload := &kafkamgmtclient.KafkaRequestPayload{}
 
-			validMarketplaceAcctIDs := FetchValidMarketplaceAccounts(orgQuota.MarketplaceQuotas, answers.Marketplace)
+	if opts.useEnterpriseCluster {
+		/*
+			if using dedicated cluster option then get the supported sizes for this cluster
+			based on the sizes it supports the user to select the one they want to use
+			then check if the size uses <= the remaining streaming units it takes up
+			on the cluster
+		*/
+		cluster, err := getClusterDetails(opts.f, opts.selectedCluster.GetId())
+		if err != nil {
+			return nil, err
+		}
 
-			if len(validMarketplaceAcctIDs) == 1 {
-				answers.MarketplaceAcctID = validMarketplaceAcctIDs[0]
-			} else {
-				marketplaceAccountPrompt := &survey.Select{
-					Message: f.Localizer.MustLocalize("kafka.create.input.marketplaceAccountID.message"),
-					Options: validMarketplaceAcctIDs,
-				}
-				err = survey.AskOne(marketplaceAccountPrompt, &answers.MarketplaceAcctID)
-				if err != nil {
-					return nil, err
-				}
-			}
+		supportedInstanceTypes := cluster.GetSupportedInstanceTypes()
+		supportedKafkaTypes := supportedInstanceTypes.GetInstanceTypes()
+
+		// right now enterprise clusters only support one type of kafka, so we are hard
+		// checking this, this assumption may change in the future
+		if len(supportedKafkaTypes) != 1 {
+			return nil, fmt.Errorf("expected one supported instance kafka type")
 		}
-	}
 
-	marketplaceInfo := accountmgmtutil.MarketplaceInfo{
-		BillingModel:   answers.BillingModel,
-		Provider:       answers.Marketplace,
-		CloudAccountID: answers.MarketplaceAcctID,
-	}
+		sizes = supportedKafkaTypes[0].Sizes
+
+		index, err := promptUserForSizes(opts.f, &sizes)
+		if err != nil {
+			return nil, err
+		}
+
+		selectedSize := sizes[index]
+
+		if selectedSize.GetCapacityConsumed() > cluster.CapacityInformation.RemainingKafkaStreamingUnits {
+			return nil, opts.f.Localizer.MustLocalizeError("kafka.create.cluster.error.noEnoughCapacity", localize.NewEntry("Size", selectedSize.GetId()))
+		}
+
+		answers.Size = selectedSize.GetId()
+
+		if len(orgQuota.EnterpriseQuotas) == 0 {
+			return nil, fmt.Errorf("No enterprise quota available")
+		}
```

Comment:
This is not checking if the user has remaining quota, it just check if the user at some point had enterprise quota. 
We should improve this onto checking for `remaining quota >= size.quota_consumed`.

Reply:
but it is getting it from here https://github.com/dimakis/app-services-cli/blob/d9826b3b71bf87661a7a5cd04d842d50029dfb68/pkg/shared/accountmgmtutil/ams.go#L111
this is where the quota is checked. makes sense?

Reply:
and it is called in create here: https://github.com/dimakis/app-services-cli/blob/d9826b3b71bf87661a7a5cd04d842d50029dfb68/pkg/cmd/kafka/create/create.go#L550

Reply:
All that is fine, but what's missing is checking that the user has truly remaining quota. I'll explain myself

> but it is getting it from here https://github.com/dimakis/app-services-cli/blob/d9826b3b71bf87661a7a5cd04d842d50029dfb68/pkg/shared/accountmgmtutil/ams.go#L111
this is where the quota is checked. makes sense?

It could be that in the `remainingQuota` is `0`. I do not see anywhere in the code where we check for this and we'll endup attempting to create a Kafka even if there is no remaining quota.

Reply:
Done